From 026a8c44d332b3595814ce0aceba255467cd7b6d Mon Sep 17 00:00:00 2001
From: Angelo Compagnucci <angelo.compagnucci@gmail.com>
Date: Sat, 5 Sep 2015 08:57:21 +0200
Subject: [PATCH] eglib: checking for locale_charset function

This patch checks if locale_charset function is availabe in
libiconv or libcharset and changes the linking options accordingly.

Signed-off-by: Angelo Compagnucci <angelo.compagnucci@gmail.com>
---
 eglib/configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/eglib/configure.ac b/eglib/configure.ac
index 9d094ea..5ea220e 100644
--- a/eglib/configure.ac
+++ b/eglib/configure.ac
@@ -182,6 +182,7 @@ fi
 AC_SUBST(G_HAVE_ISO_VARARGS)
 
 AC_CHECK_HEADERS(getopt.h sys/select.h sys/time.h sys/wait.h pwd.h langinfo.h iconv.h localcharset.h sys/types.h sys/resource.h)
+AC_CHECK_LIB([iconv], [locale_charset],[],[AC_CHECK_LIB([charset], [locale_charset],[LIBS+="-liconv -lcharset"])])
 AC_CHECK_HEADER(alloca.h, [HAVE_ALLOCA_H=1], [HAVE_ALLOCA_H=0])
 AC_SUBST(HAVE_ALLOCA_H)
 
-- 
1.9.1

